### PR TITLE
SDP 1687 update wallet patch endpoint

### PR DESCRIPTION
**Pull request by dpohr**
### What

Extended the `PATCH /wallets/:wallet_id` endpoint to support updating all wallet fields (name, homepage, sep_10_client_domain, deep_link_schema, and assets) instead of just the enabled status.

### Why

The existing PATCH endpoint only supported toggling the enabled/disabled status of wallets, which was insufficient for comprehensive wallet management.
In case of any changes the only way was to do it is via direct database manipulations, with these PR we will reduce direct DB interactions and extend capabilities of the API

### Known limitations

Depends on https://github.com/stellar/stellar-disbursement-platform-backend/pull/696

- [x] Title follows `SDP-1234: Add new feature` or `Chore: Refactor package xyz` format. The Jira ticket code was included if available.
- [x] PR has a focused scope and doesn't mix features with refactoring
- [x] Tests are included (if applicable)
- [x] `CHANGELOG.md` is updated (if applicable)
- [x] CONFIG/SECRETS changes are updated in helmcharts and deployments (if applicable)
- [x] Preview deployment works as expected
- [ ] Ready for production

### CODE REVIEW

**Review by marwen-abid — APPROVED** [review]
LGTM! Plz feel free to merge after addressing the comments.

File: internal/serve/httphandler/wallets_handler_test.go [hunk 1]
```diff
@@ -647,106 +640,306 @@ func Test_WalletsHandlerDeleteWallet(t *testing.T) {
...
+			}`,
+			walletIDFn:     func(wallet *data.Wallet) string { return wallet.ID },
+			expectedStatus: http.StatusOK,
+			validateResult: func(t *testing.T, wallet *data.Wallet, originalWallet *data.Wallet) {
+				assert.Equal(t, "Rylanor's Memorial Fund", wallet.Name)
+				assert.Len(t, wallet.Assets, 2)
+			},
+		},
+		{
+			name: "🔴 fails when no fields provided",
```
Comment (philipliu) [inline internal/serve/httphandler/wallets_handler_test.go:832]: Should we include a test that checks for whitespace trimming?

**Review by philipliu — APPROVED** [review]
